test/smoke/src/main.ts

```diff
@@ -417,25 +425,26 @@ function createOptions(): ApplicationOptions {
 		loggers.push(new ConsoleLogger());
 	}
 
-	let log: string | undefined = undefined;
+	// Prepare logs root path
+	fs.rmSync(logsRootPath, { recursive: true, force: true, maxRetries: 3 });
+	mkdirp.sync(logsRootPath);
 
-	if (opts.log) {
-		loggers.push(new FileLogger(opts.log));
-		log = 'trace';
-	}
+	// Always log to log file
+	loggers.push(new FileLogger(path.join(logsRootPath, 'smoke-test-runner.log')));
 
 	return {
 		quality,
 		codePath: opts.build,
 		workspacePath,
 		userDataDir,
-		extensionsPath,
+		extensionsPath: extensionsPath ?? '', // {{SQL CARBON EDIT}} null coalescing
```

Comment:
Please make sure any carbon edit comments explain why this is necessary - we can already see that we're null coalescing the value but why this is necessary isn't clear.

Reply:
I was having a TypeScript typing issue. The issue was: type 'string | undefined' cannot be assigned to type 'string'. Adding null coalescing was how that was resolved.